### PR TITLE
Fix clippy lints, improve Travis

### PR DESCRIPTION
**Pull request by pzmarzly**

### CODE REVIEW

File: components/index_client/src/seq_friends.rs [hunk 1]
```diff
@@ -49,7 +49,8 @@ async fn seq_friends_loop(
                 let _ = response_sender.send(());
             }
             SeqFriendsRequest::ResetCountdown(response_sender) => {
-                let _ = response_sender.send(seq_friends.reset_countdown());
```
Comment (realcr) [inline components/index_client/src/seq_friends.rs]: @pzmarzly :  This one is really strange. I wonder how it happened. Maybe reset_countdown() returned something in the past? It's pretty cool clippy caught this one.

File: components/proto/src/file/app.rs [hunk 2]
```diff
@@ -85,7 +85,7 @@ pub fn store_trusted_app_to_file(
     let data = toml::to_string(&trusted_app_file)?;
 
     let mut file = File::create(path)?;
-    file.write(&data.as_bytes())?;
+    file.write_all(&data.as_bytes())?;
```
Comment (realcr) [inline components/proto/src/file/app.rs]: @pzmarzly: Saved my life here!

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 language: rust
 dist: trusty
 sudo: required
+git:
```
[the remark this answers is not in the document]
Comment (pzmarzly) [reply .travis.yml]: By default Travis downloads last 50 commits. I don't think we need any, just the files, so downloading only the latest commit + files should be a bit faster.

File: travis/trusty/after-success.sh [hunk 4]
```diff
@@ -20,3 +20,9 @@ if [[ "$KCOV" == "1" ]]; then
     # Automatically reads from CODECOV_TOKEN environment variable:
     bash <(curl -s https://codecov.io/bash)
 fi
+
+# Install cargo-sweep. If installed, cargo failure will be ignored.
```
[the remark this answers is not in the document]
Comment (pzmarzly) [reply travis/trusty/after-success.sh]: Cargo doesn't retrigger builds of `~/.cargo/bin` utils unless `-f` flag is specified. cargo-sweep is a "dumb" utility that compiles on stable, so hopefully we won't have any issues with it. I am going to change its config a bit, though.

File: travis/trusty/before-script.sh [hunk 5]
```diff
@@ -8,16 +8,12 @@ elif [[ ! -d ".cargo" ]]; then
...
-rm -rf $HOME/.cargo/registry
-
 echo "[target.$TARGET]" > .cargo/config
 echo "linker= \"$CC\"" >> .cargo/config
 
-# Install clippy, according to:
-# https://internals.rust-lang.org/t/clippy-is-available-as-a-rustup-component/7967
 rustup update
 rustup component add clippy-preview
+rustup component add rustfmt
```
[the remark this answers is not in the document]
Comment (pzmarzly) [reply travis/trusty/before-script.sh]: I'm not sure either, but hopefully TCP socket reusing will speed up CI by 0.1 second or something

File: .travis.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (realcr) [reply .travis.yml]: Sounds good to me. I wonder why they have the last 50 commits default. I can't think of a good reason.

File: travis/trusty/post/cargo-sweep.sh [hunk 6]
```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env bash
 
 # Install cargo-sweep. If installed, cargo failure will be ignored.
+# TODO: parse ~/.cargo/.crates.toml to determine version
+#       maybe use cargo-update library for that
 cargo install cargo-sweep --vers 0.4.1 || true
 
 # cargo-sweep produces a lot of output
-cargo sweep . -t 30 > /dev/null
+cargo sweep --toolchains $TRAVIS_RUST_VERSION > /dev/null
```
Comment (realcr) [inline travis/trusty/post/cargo-sweep.sh]: Is it possible that TRAVIS_RUST_VERSION will be different from our pinned rustc nightly version? If so, is this the desired behaviour?

Comment (pzmarzly) [reply travis/trusty/post/cargo-sweep.sh]: The desired behavior is to run `cargo sweep --toolchains nightly-2019-03-20`, so everything created by other toolchains will be purged from cache (also, everything older than 30 days made by `nightly-2019-03-20`, but that's default behavior).

I hope `TRAVIS_RUST_VERSION` is equal to the pinned version, I will add `echo` just to be sure (cargo-sweep returns 0 even on error, I will report that as a bug)

Comment (realcr) [reply travis/trusty/post/cargo-sweep.sh]: I have just seen the print output, I'm convinced (:

```bash
$ travis/trusty/post/cargo-sweep.sh
    Updating crates.io index
  Installing cargo-sweep v0.4.1
error: binary `cargo-sweep` already exists in destination as part of `cargo-sweep v0.4.1`
Add --force to overwrite
Rust toolchain version: nightly-2019-03-20
The command "travis/trusty/post/cargo-sweep.sh" exited with 0.
```
I wonder how it works. I was almost sure that `TRAVIS_RUST_VERSION` will be some generic rustc version like the latest nightly.

**Pull request merged**